bootstrap/bootstrap.go

```diff
@@ -0,0 +1,322 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// TODO(bentheelder): add unit tests
+
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	flag "github.com/spf13/pflag"
+)
+
+// Constant Keys for known environment variables and URLs
+const (
+	BuildEnv          string = "BUILD_NUMBER"
+	BootstrapEnv      string = "BOOTSTRAP_MIGRATION"
+	CloudSDKEnv       string = "CLOUDSDK_CONFIG"
+	GCEKeyEnv         string = "JENKINS_GCE_SSH_PRIVATE_KEY_FILE"
+	Gubernator        string = "https://k8s-gubernator.appspot.com/build"
+	HomeEnv           string = "HOME"
+	JenkinsHomeEnv    string = "JENKINS_HOME"
+	JobEnv            string = "JOB_NAME"
+	NodeEnv           string = "NODE_NAME"
+	ServiceAccountEnv string = "GOOGLE_APPLICATION_CREDENTIALS"
+	WorkspaceEnv      string = "WORKSPACE"
+	GCSArtifactsEnv   string = "GCS_ARTIFACTS_DIR"
+)
+
+// Args contains all of the (parsed) command line arguments for bootstrap
+// NOTE: Repo should be further parsed by ParseRepos
+type Args struct {
+	Root           string
+	Timeout        int32
```

Comment:
`time.Duration` probably

Reply:
we can convert to one later on in the code, I wanted this to parse the args the same way the python one does.

Reply:
I think in the future when we switch all of the --timeout args to use a go time.Duration string but right now we have some unit tests around assuming this is an integer number of minutes